src/geometry_base/kd_tree_strategy.cc

```diff
@@ -0,0 +1,143 @@
+                                                         lane.first};
+        points.push_back(point);
+      }
+    }
+  }
+  kd_tree_ = (std::make_unique<math::KDTree3D<MaliputPoint>>(std::move(points)));
+}
+
+maliput::api::LaneId KDTreeStrategy::do_closest_lane(const maliput::math::Vector3& point) const {
+  // Given that this method should actually obtain the lanes within a tolerance radius. Afterwards, the lane
+  // could be obtained using the point being closer to the centerline.
+  const api::InertialPosition& inertial_position{point.x(), point.y(), point.z()};
+  const auto maliput_point = kd_tree_->Nearest(point);
+  double radius = std::sqrt(pow(point.x() - maliput_point.x(), 2) + pow(point.y() - maliput_point.y(), 2) +
+                            pow(point.z() - maliput_point.z(), 2)) +
+                  0.02;
+  const auto lane_ids = do_closest_lanes(point, radius);
+
+  auto lane_id_result = maliput_point.lane_id();
+  auto lane_position_result = rg_->ById().GetLane(lane_id_result)->ToLanePosition(inertial_position);
+  double min_distance = lane_position_result.distance;
+  const static double KLineraTolerance{1e-12};  // [m]
+
+  for (const auto& current_lane : lane_ids) {
+    const auto current_lane_position = rg_->ById().GetLane(maliput_point.lane_id())->ToLanePosition(inertial_position);
+    if (current_lane_position.distance - min_distance > KLineraTolerance) {
+      min_distance = current_lane_position.distance;
+      lane_position_result = current_lane_position;
+      lane_id_result = current_lane;
+      continue;
+    }
+    if (current_lane_position.distance - min_distance <= -KLineraTolerance) {
+      continue;
+    }
+    if (std::abs(current_lane_position.lane_position.r()) - std::abs(lane_position_result.lane_position.r()) <
+        -KLineraTolerance) {
+      min_distance = current_lane_position.distance;
+      lane_position_result = current_lane_position;
+      lane_id_result = current_lane;
+      continue;
+    }
+    if (std::abs(current_lane_position.lane_position.r()) - std::abs(lane_position_result.lane_position.r()) >
+        KLineraTolerance) {
+      continue;
+    }
+  }
+  return lane_id_result;
+}
+
+std::set<maliput::api::LaneId> KDTreeStrategy::do_closest_lanes(const maliput::math::Vector3& point,
+                                                                double distance) const {
+  const maliput::math::Vector3 min_corner{point.x() - distance, point.y() - distance, point.z() - distance};
+  const maliput::math::Vector3 max_corner{point.x() + distance, point.y() + distance, point.z() + distance};
+  const math::AxisAlignedBox& search_region{min_corner, max_corner};
+  const auto maliput_points = kd_tree_->RangeSearch(search_region);
+  std::set<maliput::api::LaneId> maliput_lanes;
+  for (const auto& current_point : maliput_points) {
+    maliput_lanes.insert(current_point->lane_id());
+  }
```

Comment:
Please, use std::transform for this.

Reply:
For reference: https://en.cppreference.com/w/cpp/algorithm/transform

Reply:
I tried using the std::transform for this, however, I encountered some problems because `maliput::api::LaneId` is a class that cannot be copied nor moved. I left a comment with the code I was using, if you have any idea that could work as a workaround and using std::transform I'll try it. I'll attach the error I was getting 
![image](https://user-images.githubusercontent.com/51831786/189957922-7fbda35d-56c9-4509-845f-ffb2d451df8d.png)